controllers/argocd/keycloak.go

```diff
@@ -1404,49 +1409,56 @@ func (r *ReconcileArgoCD) reconcileKeycloakForOpenShift(cr *argoprojv1a1.ArgoCD)
 		// keycloakRouteURL is used to update the OIDC configuration for ArgoCD.
 		keycloakRouteURL := cfg.KeycloakURL
 
-		// Create a keycloak realm and publish.
-		response, err := createRealm(cfg)
-		if err != nil {
-			log.Error(err, fmt.Sprintf("Failed posting keycloak realm configuration for ArgoCD %s in namespace %s",
-				cr.Name, cr.Namespace))
-			return err
-		}
-
-		if response == successResponse {
-			log.Info(fmt.Sprintf("Successfully created keycloak realm for ArgoCD %s in namespace %s",
-				cr.Name, cr.Namespace))
+		// If Keycloak deployment exists and a realm is already created for ArgoCD, Do not create a new one.
+		if existingDC.Annotations["argocd.argoproj.io/realm-created"] == "false" {
 
-			// TODO: Remove the deleteOAuthClient invocation once the issue is resolved.
-			// OAuthClient configuration does not get deleted from previous instances occasionally.
-			// It is safe to delete before updating the OIDC config.
-			// https://github.com/openshift/client-go/issues/209
-			err = deleteOAuthClient(cr)
+			// Create a keycloak realm and publish.
+			response, err := createRealm(cfg)
 			if err != nil {
+				log.Error(err, fmt.Sprintf("Failed posting keycloak realm configuration for ArgoCD %s in namespace %s",
+					cr.Name, cr.Namespace))
 				return err
 			}
 
-			// Update Realm creation. This will avoid posting of realm configuration on further reconciliations.
-			err = r.Client.Get(context.TODO(), types.NamespacedName{Name: existingDC.Name, Namespace: existingDC.Namespace}, existingDC)
-			if err != nil {
-				return err
-			}
+			if response == successResponse {
+				log.Info(fmt.Sprintf("Successfully created keycloak realm for ArgoCD %s in namespace %s",
+					cr.Name, cr.Namespace))
 
-			existingDC.Annotations["argocd.argoproj.io/realm-created"] = "true"
-			err = retry.RetryOnConflict(retry.DefaultBackoff, func() error {
-				return r.Client.Update(context.TODO(), existingDC)
-			})
+				// TODO: Remove the deleteOAuthClient invocation once the issue is resolved.
+				// OAuthClient configuration does not get deleted from previous instances occasionally.
+				// It is safe to delete before updating the OIDC config.
+				// https://github.com/openshift/client-go/issues/209
+				err = deleteOAuthClient(cr)
+				if err != nil {
+					return err
+				}
 
-			if err != nil {
-				return err
-			}
+				// Update Realm creation. This will avoid posting of realm configuration on further reconciliations.
+				err = r.Client.Get(context.TODO(), types.NamespacedName{Name: existingDC.Name, Namespace: existingDC.Namespace}, existingDC)
+				if err != nil {
+					return err
+				}
```

Comment:
@iam-veeramalla any particular reason for introducing this get call here? seems like we would already have access to the existingDC resource after getting it on L 1381

Reply:
@jaideepr97 I did not make any changes to the existing code.

Reply:
That call has been added in this PR though, since it shows up in green

Reply:
Hey, its issue with git, please check the diff :)